config/v3_0/types/systemd_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/coreos/ignition/v2/config/shared/errors"
+	"github.com/coreos/ignition/v2/config/util"
+	"github.com/coreos/vcontext/path"
+	"github.com/coreos/vcontext/report"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSystemdValidate(t *testing.T) {
+	tests := []struct {
+		in  Systemd
+		out error
+	}{
+		{
+			Systemd{
+				[]Unit{
+					{Name: "test@.service", Contents: util.StrToPtr("[Foo]\nQux=Bar")},
+					{Name: "test@foo.service", Enabled: util.BoolToPtr(true)},
```

Comment:
Let's also add:
- An enabled instance unit referring to a missing instantiable unit
- An enabled instance unit referring to an instantiable unit with no contents